src/segment_reshape/map_tool/segment_reshape_tool.py

```diff
@@ -158,45 +170,64 @@ def _handle_mouse_click_event(
                 success=True,
             )
 
+    def _handle_key_event(self, key: Qt.Key) -> None:
+        if self.reshape_mode is True:
+            if key == Qt.Key_Escape:
+                self.new_segment_rubber_band.reset()
+                self.temporary_new_segment_rubber_band.reset()
+                self._change_to_pick_location_mode()
+                return
```

Comment:
This could be extracted to own method for example `self._abort_reshape()` for clarity.